static-content-hosting/README.md

```diff
@@ -0,0 +1,136 @@
+
+### Steps
+
+1. Clone/download this repo locally.
+
+   ```bash
+   git clone https://github.com/mspnp/cloud-design-patterns.git
+   cd cloud-design-patterns/static-content-hosting
+   ```
+
+1. Log into Azure and create an empty resource group.
+
+   ```azurecli
+   az login
+   az account set -s <Name or ID of subscription>
+   ```
+
+1. Create an empty resource group
+
+   ```bash
+   # [This takes less than one minute to run.]
+   az group create -n rg-hosting-static-content -l eastus2
+   ```
+
+1. Control Plane - Deploy a new Azure storage account
+
+   ```bash
+   STORAGE_ACCOUNT_NAME="stvaletblobs$(LC_ALL=C tr -dc 'a-z0-9' < /dev/urandom | fold -w 7 | head -n 1)"
+
+   # [This takes about one minute to run.]
+   az deployment group create -n deploy-hosting-static-content -f bicep/main.bicep -g rg-hosting-static-content -p storageAccountName=$STORAGE_ACCOUNT_NAME
+   ```
+
+   > :warning: for those using devcontainers and running on arm platform, consider `az bicep uninstall && az bicep install --target-platform linux-arm64`
+
+1. [Control Plane] RBAC your user with Storage Blob Data Contributor. This is required to be able to upload files to the recently created storage account
+
+   ```bash
+   ROLEASSIGNMENT_TO_UPLOAD_CONTENT=$(az role assignment create --role ba92f5b4-2d11-453d-a403-e96b0029c9fe --assignee-principal-type user --assignee-object-id $(az ad signed-in-user show --query 'id' -o tsv) --scope $(az storage account show -g rg-hosting-static-content -n $STORAGE_ACCOUNT_NAME --query 'id' -o tsv) --query 'id' -o tsv)
+   echo ROLEASSIGNMENT_TO_UPLOAD_CONTENT: $ROLEASSIGNMENT_TO_UPLOAD_CONTENT
+   ```
+
+1. [Data Plane] Enable static website and upload content
+
+   ```bash
+   # Enables static website
+   az storage blob service-properties update --account-name $STORAGE_ACCOUNT_NAME --static-website --404-document 404.html --index-document index.html --auth-mode login
+
+   # upload more files
+   az storage blob upload-batch -s "./src/static" --destination "\$web" --account-name $STORAGE_ACCOUNT_NAME --pattern "*.html" --content-type "text/html" --content-cache max-age=3600 --auth-mode login
+   az storage blob upload-batch -s "./src/static" --destination "\$web" --account-name $STORAGE_ACCOUNT_NAME --pattern "*.css" --content-type "text/css" --content-cache max-age=3600 --auth-mode login
+   ```
+
+1. Obtain the public url
+
+   ```bash
+   # Retrieve the static website endpoint
+   export STATIC_WEBSITE_URL=$(az storage account show -n $STORAGE_ACCOUNT_NAME -g rg-hosting-static-content  --query primaryEndpoints.web --output tsv)
+   ```
+
+1. Generate a _index.html_ document file for validation purposes
+
+   ```bash
+   cat > src/index.html << EOF
+   <!DOCTYPE html>
+   <html>
+   <head>
+       <link href="${STATIC_WEBSITE_URL}style.css" rel="stylesheet" type="text/css" data-preload="true"/>
+   </head>
+   <body>
+       <h1>Welcome to contoso.com</h1>
+   </body>
+   </html>
+   EOF
```

Comment:
Any thoughts on just doing a `sed` in a file that already exists?

Reply:
just replaced with `sed` approach instead  @ckittel 

done | addressed from 6a31de6d343937ae57c3c15a3e8b6bdc7349376a